arches_lingo/src/arches_lingo/components/generic/resource-instance-relationships/ResourceInstanceRelationshipsEditor.vue

```diff
@@ -1,23 +1,71 @@
 <script setup lang="ts">
-import { computed, toRef, ref, watch } from "vue";
+import { computed, onMounted, ref, toRef, watch } from "vue";
 import MultiSelect from "primevue/multiselect";
-import type { ResourceInstanceReference } from "@/arches_lingo/types";
+import Button from "primevue/button";
+import Dialog from "primevue/dialog";
+import type {
+    GraphInfo,
+    NewResourceInstance,
+    ResourceInstanceReference,
+    ResourceInstanceResult,
+} from "@/arches_lingo/types";
 import { useGettext } from "vue3-gettext";
+import { fetchRelatableResources } from "@/arches_lingo/api.ts";
+import {
+    UPDATED,
+    CREATE_NEW_RESOURCE,
+    ERROR,
+    DEFAULT_ERROR_TOAST_LIFE,
+} from "@/arches_lingo/constants.ts";
+import type { VirtualScrollerLazyEvent } from "primevue/virtualscroller";
+import { useToast } from "primevue/usetoast";
 
+const ITEM_HEIGHT = 38;
+const RESULTS_PER_PAGE = 25;
```

Comment:
nit: since this will be used as a template for genericizing should these be passed in as props?

Reply:
this is fair